wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/environment/server/EnvironmentParser.java

```diff
@@ -106,20 +110,55 @@ public ComposeEnvironmentImpl parse(Environment environment) throws IllegalArgum
         return composeEnvironment;
     }
 
-    private ComposeEnvironmentImpl parseCompose(EnvironmentRecipe recipe) throws ServerException {
+    private CheServicesEnvironmentImpl parseCompose(EnvironmentRecipe recipe) throws ServerException {
         String recipeContent = getContentOfRecipe(recipe);
 
-        return composeFileParser.parse(recipeContent, recipe.getContentType());
+        ComposeEnvironmentImpl composeEnvironment = composeFileParser.parse(recipeContent, recipe.getContentType());
+
+        return asCheEnvironment(composeEnvironment);
+    }
+
+    private CheServicesEnvironmentImpl asCheEnvironment(ComposeEnvironmentImpl composeEnvironment) {
+        Map<String, CheServiceImpl> services = Maps.newHashMapWithExpectedSize(composeEnvironment.getServices().size());
+        for (Map.Entry<String, ComposeServiceImpl> composeServiceEntry : composeEnvironment.getServices()
+                                                                                           .entrySet()) {
+            ComposeServiceImpl service = composeServiceEntry.getValue();
+
+            CheServiceBuildContextImpl buildContext = null;
+            if (service.getBuild() != null) {
+                buildContext = new CheServiceBuildContextImpl().withContext(service.getBuild().getContext())
+                                                               .withDockerfilePath(service.getBuild().getDockerfile());
+            }
+
+            CheServiceImpl cheService = new CheServiceImpl().withBuild(buildContext)
```

Comment:
Not sure this class needs such a builder. Notice that I build one class from another (very similar) class.